src/snowflake/snowpark/profiler.py

```diff
@@ -0,0 +1,207 @@
+
+    def _set_targeted_stage(self):
+        self.session.sql(self.set_targeted_stage_sql).collect()
+
+    def _set_active_profiler(self):
+        self.session.sql(self.set_active_profiler_sql).collect()
+
+    def enable_profiler(self):
+        """
+        Enable profiler. Profiles will be generated until profiler is disabled.
+        """
+        self.session.sql(self.enable_profiler_sql).collect()
+
+    def disable_profiler(self):
+        """
+        Disable profiler.
+        """
+        self.session.sql(self.disable_profiler_sql).collect()
+
+    def _get_last_query_id(self):
+        pattern = r"WITH\s+.*?\s+AS\s+PROCEDURE\s+.*?\s+CALL\s+.*"
+        for query in self.query_history.queries[::-1]:
+            if query.sql_text.startswith("CALL") or re.match(
+                pattern, query.sql_text, re.DOTALL
+            ):
+                return query.query_id
+        return None
+
+    def show_profiles(self) -> str:
+        """
+        Return and show the profiles of last executed stored procedure.
+
+        Note:
+            This function must be called right after the execution of stored procedure you want to profile.
+        """
+        query_id = self._get_last_query_id()
+        sql = f"select snowflake.core.get_python_profiler_output('{query_id}')"
+        res = self.session.sql(sql).collect()
+        print(res[0][0])  # noqa: T201: we need to print here.
+        return res[0][0]
+
+    def dump_profiles(self, dst_file: str):
+        """
+        Write the profiles of last executed stored procedure to given file.
+
+        Note:
+            This function must be called right after the execution of stored procedure you want to profile.
+
+        Args:
+            dst_file: String of file name that you want to store the profiles.
+        """
+        query_id = self._get_last_query_id()
+        sql = f"select snowflake.core.get_python_profiler_output('{query_id}')"
+        res = self.session.sql(sql).collect()
+        with open(dst_file, "w") as f:
+            f.write(str(res[0][0]))
+
+
+@contextmanager
+def profiler(
+    stage: str,
+    active_profiler: str,
+    session: "snowflake.snowpark.Session",
+    modules: Optional[List[str]] = None,
+):
+    internal_profiler = Profiler(stage, active_profiler, session)
+    session.profiler = internal_profiler
+    internal_profiler.query_history = session.query_history()
+    modules = modules or []
+    try:
+        # create stage if not exist
+        if (
+            len(session.sql(f"show stages like '{internal_profiler.stage}'").collect())
+            == 0
+            and len(
+                session.sql(
+                    f"show stages like '{internal_profiler.stage.split('.')[-1]}'"
+                ).collect()
+            )
+            == 0
+        ):
+            session.sql(
+                f"create temp stage {internal_profiler.stage} FILE_FORMAT = (RECORD_DELIMITER = NONE FIELD_DELIMITER = NONE )"
```

Comment:
nit: You can use the SQL syntax as well
```suggestion
                f"create temp stage if not exist {internal_profiler.stage} FILE_FORMAT = (RECORD_DELIMITER = NONE FIELD_DELIMITER = NONE )"
```